src/integrationTest/java/org/opensearch/security/http/JwtAuthorizationHeaderFactory.java

```diff
@@ -0,0 +1,141 @@
+/*
+* Copyright OpenSearch Contributors
+* SPDX-License-Identifier: Apache-2.0
+*
+* The OpenSearch Contributors require contributions made to
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*
+*/
+package org.opensearch.security.http;
+
+import java.security.PrivateKey;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableMap;
+import io.jsonwebtoken.Jwts;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.message.BasicHeader;
+
+import static io.jsonwebtoken.SignatureAlgorithm.RS256;
+import static java.util.Objects.requireNonNull;
+
+class JwtAuthorizationHeaderFactory {
+	public static final String AUDIENCE = "OpenSearch";
+	public static final String ISSUER = "test-code";
+	private final PrivateKey privateKey;
+
+	private final String usernameClaimName;
+
+	private final String rolesClaimName;
+
+	private final String headerName;
+
+	public JwtAuthorizationHeaderFactory(PrivateKey privateKey, String usernameClaimName, String rolesClaimName, String headerName) {
+		this.privateKey = requireNonNull(privateKey, "Private key is required");
+		this.usernameClaimName = requireNonNull(usernameClaimName, "Username claim name is required");
+		this.rolesClaimName = requireNonNull(rolesClaimName, "Roles claim name is required.");
+		this.headerName = requireNonNull(headerName, "Header name is required");
+	}
+
+	Header validToken(String username, String...roles) {
```

Comment:
nit: This method generates a validToken correct? If so can you please rename it as such?

Reply:
And the methods below as well

Reply:
Corrected.